src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -201,11 +195,11 @@ private async Task Invoke(HttpContext httpContext, bool retry)
 
             // First check for a HMAC token and capture before the command is stripped out.
             byte[] secret = this.options.HMACSecretKey;
-            bool checkHMAC = false;
+            bool computeHMAC = false;
```

Comment:
Changing the variable name here incorrectly changes the meaning of the bool field.

Reply:
That's because the meaning _has_ changed! Instead of always requiring a valid token when a secret is configured, it's dependent on whether the `OnComputeHMACAsync` returns null or not.

Together with removing the caching, this allows you to bypass the HMAC check based on the request, e.g. whether a user is logged in.